x-pack/plugin/identity-provider/qa/idp-rest-tests/src/test/java/org/elasticsearch/xpack/idp/IdentityProviderAuthenticationIT.java

```diff
@@ -0,0 +1,240 @@
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.common.xcontent.ObjectPath;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.xpack.idp.saml.sp.SamlServiceProviderIndex;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken.basicAuthHeaderValue;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class IdentityProviderAuthenticationIT extends IdpRestTestCase {
+
+    // From build.gradle
+    private final String SP_ENTITY_ID = "ec:123456:abcdefg";
+    private final String SP_ACS = "https://sp1.test.es.elasticsearch.org/saml/acs";
+    private final String REALM_NAME = "cloud-saml";
+
+    @Before
+    public void createUsers() throws IOException {
+        setUserPassword("kibana", new SecureString("kibana".toCharArray()));
+    }
+
+    public void testRegistrationAndIdpInitiatedSso() throws Exception {
+        final Map<String, Object> request = Map.ofEntries(
+            Map.entry("name", "Test SP"),
+            Map.entry("acs", SP_ACS),
+            Map.entry("privileges", Map.ofEntries(
+                Map.entry("resource", SP_ENTITY_ID),
+                Map.entry("roles", Map.of("superuser", "role:superuser", "viewer", "role:viewer"))
+            )),
+            Map.entry("attributes", Map.ofEntries(
+                Map.entry("principal", "https://idp.test.es.elasticsearch.org/attribute/principal"),
+                Map.entry("name", "https://idp.test.es.elasticsearch.org/attribute/name"),
+                Map.entry("email", "https://idp.test.es.elasticsearch.org/attribute/email"),
+                Map.entry("roles", "https://idp.test.es.elasticsearch.org/attribute/roles")
+            )));
+        final SamlServiceProviderIndex.DocumentVersion docVersion = createServiceProvider(SP_ENTITY_ID, request);
+        checkIndexDoc(docVersion);
+        ensureGreen(SamlServiceProviderIndex.INDEX_NAME);
+        final String samlResponse = generateSamlResponse(SP_ENTITY_ID, SP_ACS, null);
+        authenticateWithSamlResponse(samlResponse, null);
+    }
+
+    public void testRegistrationAndSpInitiatedSso() throws Exception {
```

Comment:
Did you have a thought on how we would test wildcard services here?
Just another method, or switch one of these to be wildcarded rather than registered?

I don't think we want to hold up this PR for it, but we should have a follow up to add it.

Reply:
I was thinking of adding method(s) for wildcarded - will take care of it in a followup